pkg/configmap/configmap.go

```diff
@@ -0,0 +1,166 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sort"
+	"time"
+)
+
+type (
+	Service interface {
+		ListNamesAndLabels() (configMapNames, labels []string, err error)
+		List(listOptions metav1.ListOptions) (configMaps []v1.ConfigMap, err error)
+		GetUnused(namespace string, configMaps []v1.ConfigMap) (unusedConfigMaps []v1.ConfigMap, funcErr error)
+		Delete(configMaps []v1.ConfigMap)
+		FilterByTime(configMaps []v1.ConfigMap, olderThan time.Time) (filteredConfigMaps []v1.ConfigMap)
+		FilterByMaxCount(configMaps []v1.ConfigMap, keep int) (filteredConfigMaps []v1.ConfigMap)
+		Print(configMaps []v1.ConfigMap, batch bool)
+	}
+	ConfigMapsService struct {
+		client core.ConfigMapInterface
+		helper kubernetes.Kubernetes
+	}
+)
+
+// Constructor
+func NewConfigMapsService(client core.ConfigMapInterface, helper kubernetes.Kubernetes) ConfigMapsService {
+	return ConfigMapsService{
+		client: client,
+		helper: helper,
+	}
+}
+
+// ListNamesAndLabels return names and labels of Config Maps
+func (cms ConfigMapsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	configMaps, err := cms.List(metav1.ListOptions{})
+	var objectMetas []metav1.ObjectMeta
+	for _, cm := range configMaps {
+		objectMetas = append(objectMetas, cm.ObjectMeta)
+	}
+	configMapNames, labels := util.GetNamesAndLabels(objectMetas)
+	return configMapNames, labels, nil
+}
+
+// List returns a list of ConfigMaps from a namespace
+func (cms ConfigMapsService) List(listOptions metav1.ListOptions) ([]v1.ConfigMap, error) {
+	configMaps, err := cms.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return configMaps.Items, nil
+}
+
+// GetUnused return unused Config Maps
+func (cms ConfigMapsService) GetUnused(namespace string, configMaps []v1.ConfigMap) (unusedConfigMaps []v1.ConfigMap, funcErr error) {
+	var usedConfigMaps []v1.ConfigMap
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(configMaps, func(resource v1.ConfigMap) {
+
+			resourceName := resource.GetName()
+
+			if funk.Contains(usedConfigMaps, resource) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := cms.helper.ResourceContains(namespace, resourceName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedConfigMaps = append(usedConfigMaps, resource)
+			}
+		})
+	})
+
+	for _, resource := range configMaps {
+		if !funk.Contains(usedConfigMaps, resource) {
+			unusedConfigMaps = append(unusedConfigMaps, resource)
+		}
+	}
+
+	return unusedConfigMaps, funcErr
+}
+
+// Delete removes Config Maps
+func (cms ConfigMapsService) Delete(configMaps []v1.ConfigMap) {
+	for _, resource := range configMaps {
+		namespace := resource.GetNamespace()
+		name := resource.GetName()
+
+		log.Infof("Deleting config maps %s/%s", namespace, name)
```

Comment:
It may also make sense to check if batch mode is enabled. In batch mode, only the names of successful deletions would be printed, to make shell pipings easier (`seiso -b configmaps ... --delete > list-of-deleted-stuff.txt`)